File: app/src/main/java/com/akhbulatov/githubsample/ui/main/favoritesroot/favorites/FavoritesAdapter.java
package com.akhbulatov.githubsample.ui.main.favoritesroot.favorites;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.akhbulatov.githubsample.R;
import com.akhbulatov.githubsample.models.UserDetails;
import com.squareup.picasso.Picasso;

import java.util.List;

public final class FavoritesAdapter extends RecyclerView.Adapter<FavoritesAdapter.FavoriteViewHolder> {

    @NonNull private List<UserDetails> favorites;

    FavoritesAdapter(@NonNull List<UserDetails> favorites) {
        this.favorites = favorites;
    }

    @NonNull
    @Override
    public FavoriteViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemView = LayoutInflater
                .from(parent.getContext())
                .inflate(R.layout.item_favorite, parent, false);
        return new FavoriteViewHolder((itemView));
    }

    @Override public void onBindViewHolder(@NonNull FavoriteViewHolder holder, int position) {
        holder.bind(favorites.get(position));
    }

    @Override public int getItemCount() {
        return favorites.size();
    }

    static class FavoriteViewHolder extends RecyclerView.ViewHolder {

        private ImageView avatarImageView;
        private TextView loginTextView;
        private TextView nameTextView;

        FavoriteViewHolder(@NonNull View itemView) {
            super(itemView);
            avatarImageView = itemView.findViewById(R.id.avatarImageView);
            loginTextView = itemView.findViewById(R.id.loginTextView);
            nameTextView = itemView.findViewById(R.id.nameTextView);
        }

        void bind(@NonNull UserDetails favorites) {
            Picasso.get()
                    .load(favorites.avatarUrl)
                    .placeholder(R.drawable.ic_account_circle_black)
                    .into(avatarImageView);
            loginTextView.setText(favorites.login);
            nameTextView.setText(favorites.name);
        }
    }
}

File: app/src/main/java/com/akhbulatov/archsample/ui/main/favorites/FavoritesAdapter.kt
package com.akhbulatov.archsample.ui.main.favorites

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.akhbulatov.archsample.R
import com.akhbulatov.archsample.models.UserDetails
import com.squareup.picasso.Picasso
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.item_favorite.*

class FavoritesAdapter(
    private val favorites: List<UserDetails>
) : RecyclerView.Adapter<FavoritesAdapter.FavoriteViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FavoriteViewHolder {
        val itemView = LayoutInflater
            .from(parent.context)
            .inflate(R.layout.item_favorite, parent, false)
        return FavoriteViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: FavoriteViewHolder, position: Int) =
        holder.bind(favorites[position])

    override fun getItemCount(): Int = favorites.size

    class FavoriteViewHolder(override val containerView: View) :
        RecyclerView.ViewHolder(containerView), LayoutContainer {

        fun bind(favorites: UserDetails) {
            Picasso.get()
                .load(favorites.avatarUrl)
                .placeholder(R.drawable.ic_account_circle_black)
                .into(avatarImageView)
            loginTextView.text = favorites.login
            nameTextView.text = favorites.name
        }
    }
}